src/TableFunctions/TableFunctionMerge.cpp

```diff
@@ -49,58 +49,66 @@ void TableFunctionMerge::parseArguments(const ASTPtr & ast_function, ContextPtr
             " - name of source database and regexp for table names.",
             ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH);
 
-    args[0] = evaluateConstantExpressionForDatabaseName(args[0], context);
-    args[1] = evaluateConstantExpressionAsLiteral(args[1], context);
+    auto [is_regexp, source_database_name_or_regexp_] = evaluateDatabaseNameForMergeEngine(args[0], context);
```

Comment:
The same is here - looks like this is the reason for ` DB::Exception: Error while executing table function merge. Neither no one database matches regular expression default nor in database matches default no one table matches regular expression`, becuase `currentDatabase()` needs to be substituted to the needed database in ast (because ` executeTableFunction` uses a modified ast afterwards) and after this change it doesn't.

Reply:
Got it.